arxiv/how_can_we_be_so_dense/README.md

```diff
@@ -0,0 +1,188 @@
+# How Can We Be So Dense? The Benefits of Using Highly Sparse Representations
+
+## Abstract
+
+Most artificial networks today rely on dense representations, whereas biological networks rely on sparse representations. In this paper we show how sparse representations can be more robust to noise and interference, as long as the underlying dimensionality is sufficiently high. A key intuition that we develop is that the ratio of the operable volume around a sparse vector divided by the volume of the representational space decreases exponentially with dimensionality. We then analyze computationally efficient sparse networks containing both sparse weights and activations. Simulations on MNIST and the Google Speech Command Dataset show that such networks demonstrate significantly improved robustness and stability compared to dense networks, while maintaining competitive accuracy. We discuss the potential benefits of sparsity on accuracy, noise robustness, hyperparameter tuning, learning speed, computational efficiency, and power requirements.
+
+## Running experiments
+
+Below are instructions for reproducing all the charts and tables presented in the paper.
+
+### Prerequisites
+
+All the scripts in this directory were implemented using [python 2.7][2]. 
+
+Once python version 2.7 is installed and configured in your system, use the following command to install all required python libraries and dependencies:
+
+```
+python setup.py --user develop
+``` 
+
+And the following script to download _Google Speech Commands Dataset_:
+```
+cd projects/speech_commands/data
+./download_speech_commands.sh
+```
+
+Alternatively, you may use [docker][3] to run the experiments in a container:
+```
+docker build -t htmpaper .
+docker run -it htmpaper /bin/bash
+```  
+
+### Training the models
+
+All experiments have a common script called `run_experiment.py` located in the root of each project used to train and test the models. Once the models are trained, you may use the other script in the specific project folder to generate the figures and data used in the paper.
```

Comment:
Should we say, "We recommend using GPU acceleration. Otherwise some of the training, particularly for Google Speech Commands, will take significantly longer."

Reply:
done